metldata/submission_registry/models.py

```diff
@@ -83,3 +100,49 @@ def current_status(self) -> SubmissionStatus:
         sorted_history = sorted(self.status_history, key=attrgetter("timestamp"))
 
         return sorted_history[-1].new_status
+
+    # pylint: disable=no-self-argument
+    @validator("accession_map")
+    def check_accession_uniqueness(
+        cls, value: dict[str, dict[str, str]]
+    ) -> dict[str, dict[str, str]]:
+        """Check that no accessions are re-used accross classes."""
+
+        total_resources = 0
+        all_accessions: set[str] = set()
+
+        for resources in value.values():
+            total_resources += len(resources)
+            all_accessions.update(set(resources.values()))
+
+        if len(all_accessions) != total_resources:
+            raise ValueError("Accessions are not unique.")
+
+        return value
+
+    # pylint: disable=no-self-argument
+    @root_validator(skip_on_failure=True)
+    def check_accession_map_sync(cls, values):
+        """Check that for all anchored resources specified in the content an entry
+        exists in the access_map."""
+
+        if not values["content"]:
+            if values["accession_map"]:
+                raise ValueError(
+                    "The accession_map is not empty, but the content is empty."
+                )
+            return values
+
+        if not values["content"].keys() == values["accession_map"].keys():
```

Comment:
```suggestion
        if values["content"].keys() != values["accession_map"].keys():
```

I wanted to suggest writing `set(values[...]) != set(values[...])` but then I noticed this also works (even with different ordering) since keys views already have set-like behavior.